glowfic-dl.py

```diff
@@ -130,11 +161,75 @@ def __init__(self, stamped_urls: list[StampedURL], title: str):
         self.last_update = max((stamped_url.stamp for stamped_url in stamped_urls))
 
 
+#################
+##   Helpers   ##
+#################
+
+
+def make_filename_valid_for_epub3(filename: str) -> str:
```

Comment:
Two concerns here:

* Is this something ebooklib handles? Probably worth checking if you haven't.
* Could you write a few unit tests for this? I don't want to be too much of a stickler on a small project like this but this function is pretty complicated and also rarely exercised.

Reply:
ebooklib seems not to quite do _nothing_ in this field, but neither does what it _does_ do produce well-formed filenames. It seems to, for whatever reason, truncate filenames down to 259 characters (or possibly bytes, I only tested with single-byte characters) instead of 255? And it does nothing to filter out disallowed characters such as DEL. So the extra filtering step definitely does seem to be needed, I'll get started on some basic tests.

Reply:
Basic unit testing is now in place! Setting it up required refactoring the code from one module into several, in order to avoid difficulties importing `glowfic-dl.py` into the test file in light of the hyphen in its name, so that's now done too.

(Workarounds in theory could have been used to import it even given the hyphen's presence—`importlib.import_module`, for instance—but I suspect they'd have ended up as technical debt in the long run, whereas splitting into modules is probably healthy for the long-term shape of the project _anyway_ as it expands.)